submissions/the-darkesttt/js-post-oop/index.js

```diff
@@ -0,0 +1,92 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/the-darkesttt/a-tiny-JS-world
+   Web app: https://the-darkesttt.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+
+class Creature {
+
+   constructor( name, gender, saying, species) {
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+      this.species = species;
+   }
+
+   showValues() {
+      const valueKeys = [
+         'species',
+         'name',
+         'gender',
+         'saying'
+      ];
+      const inhabitantValues = valueKeys.map((key) => {
+         return `${this[key]}`;
+      });
+      return inhabitantValues;
+   }
+}
+
+class Animal extends Creature {
+   constructor(name, gender, saying, paws, species) {
+      super(name, gender, saying, species);
+      this.paws = paws;
+   }
+
+   showValues() {
+      const valueKeys = [
+         'paws'
+      ];
+      let inhabitantValues = valueKeys.map((key) => {
+         return `${this[key]}`;
+      });
+      let inhabitantValuesNew = super.showValues().concat(inhabitantValues);
+      return print(inhabitantValuesNew.join('; '));
```

Comment:
What if a subclass will want its own `showValues`? Adding `print` at this point will disable further extension.
`return print(...)` doesn't make sense as `print` returns `undefined`.
Avoid side effects. Or put them aside until very end (to the calling code).

Also this method does two things - (1) build values for representation and (2) actual output. Separation of concerns is a thing.